Library/Homebrew/formula.rb

```diff
@@ -405,11 +414,21 @@ def linked_keg
     Pathname.new("#{HOMEBREW_LIBRARY}/LinkedKegs/#{name}")
   end
 
+  def latest_head_prefix
+    head_prefixes = installed_prefixes.select do |p|
+      p.basename.to_s.start_with?("HEAD")
+    end
+
+    head_prefixes.sort_by do |p|
+      Tab.for_keg(Keg.new(p)).source_modified_time
+    end.last
```

Comment:
You can use `max_by { … }` instead of `sort_by { … }.last` if you're not actually interested in the sorted array. (Sorting the array just to discard it afterwards is more expensive and makes the purpose of the code less obvious.)